lib/main.dart

```diff
@@ -300,9 +344,48 @@ class _MyHomePageState extends State<MyHomePage> {
   }
   Future<void> _pullRefresh() async {
     analytics.logEvent(name: "PullToRefresh");
-    setState(() {
-      scriptureList = refreshScriptureList();
-    });
+    scriptureList = refreshScriptureList(currentList);
+    setState((){});
+  }
+
+  Future<void> _pushCollectionsScreen() async {
+    Navigator.of(context).push(
+      // Add lines from here...
+        MaterialPageRoute<void>(
+          builder: (context) {
+            return Scaffold(
```

Comment:
@alexdivadi  you want to extract this into another widget and put in its own file collections_screen.dart.

you can use the quickfix to extract to its own widget by hovering on scaffold.